### PR TITLE
Adds completions for powershell

### PR DESCRIPTION
**Pull request by tonysathre**

### CODE REVIEW

**Comment by tonysathre** [comment]
Tested on:

* Windows
  * PowerShell Core v7.2.1
  * PowerShell Desktop v5.1

* Linux
  * PowerShell Core 7.2.1

**Comment by tonysathre** [comment]
Do I need to add your copyright info to the `quick-lint-js.ps1` file?

File: completions/quick-lint-js.ps1 [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+              '--lsp-server',
+              '--config-file=',
+              '--stdin',
+              '--exit-fail-on=',
+              '--output-format=',
+              '--diagnostic-hyperlinks=',
+              '--vim-file-bufnr='
+            )
+
+    $opts.Where({$_ | Select-String $wordToComplete})
```
Comment (strager) [inline completions/quick-lint-js.ps1:15]: Would this accidentally match `el` -> `--help`?

**Comment by strager** [comment]
> Do I need to add your copyright info to the `quick-lint-js.ps1` file?

Yes.

File: completions/quick-lint-js.ps1 [hunk 2]
```diff
@@ -0,0 +1,16 @@
+Register-ArgumentCompleter -CommandName quick-lint-js -Native -ScriptBlock {
```
Comment (strager) [inline completions/quick-lint-js.ps1:1]: You should list this file in `completions/CMakeLists.txt` so it gets added to the Windows .zip downloadable.

**Review by strager — APPROVED** [review]

File: completions/quick-lint-js.ps1 (hunk 1 above)
Comment (tonysathre) [reply completions/quick-lint-js.ps1]: It did. This has been fixed in e8a766e58a60f0414f66d7653913950c0e1842e2.

**Comment by tonysathre** [comment]
Working fine on PowerShell Core for WIndows and Linux. For some reason PowerShell Desktop (v5.1) does not work consistently. Need to try it on another machine.

**Comment by tonysathre** [comment]
> Working fine on PowerShell Core for WIndows and Linux. For some reason PowerShell Desktop (v5.1) does not work consistently. Need to try it on another machine.

I feel like this must be a bug in .NET 4.6, or intended behavior that was changed in more recent versions.

`quick-lint-js <tab>` works
`quick-lint-js -<tab>` doesn't work
`quick-lint-js --<tab>` doesn't work
`quick-lint-js <ctrl+space>` works

I also tested this with `winget.exe`'s [completions](https://github.com/microsoft/winget-cli/blob/master/doc/Completion.md) and that had the same issue so I'm guessing this is intended behavior.

Is this satisfactory as is?

File: completions/quick-lint-js.ps1 (hunk 2 above)
Comment (tonysathre) [reply completions/quick-lint-js.ps1]: Is this correct? I haven't worked with CMake.

**Comment by strager** [comment]
> Is this satisfactory as is?

Sure. Some completion is better than none, I guess.

File: completions/CMakeLists.txt [hunk 3]
```diff
@@ -25,6 +25,11 @@ install(
   DESTINATION "${CMAKE_INSTALL_DATAROOTDIR}/fish/vendor_completions.d"
 )
 
+install(
+  FILES quick-lint-js.ps1
+  DESTINATION "${CMAKE_INSTALL_DATAROOTDIR}/powershell/completions"
+)
```
Comment (strager) [inline completions/CMakeLists.txt:31]: :+1:

**Review by strager — APPROVED** [review]

**Comment by strager** [comment]
Please rebase/merge. GitHub thinks there are conflicts.

Then, let me know when I should land your patches.

**Comment by strager** [comment]
Landed as commit 88b7e65511913590ba074f108a4b9ec481256cdc. (GitHub wasn't cooperating.)

**Pull request closed without merging**